zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -360,14 +367,14 @@ where
                 version: ZCASH_BLOCK_VERSION,
 
                 previous_block_hash: GetBlockHash(tip_hash),
-                block_commitments_hash: [0; 32].into(),
-                light_client_root_hash: [0; 32].into(),
-                final_sapling_root_hash: [0; 32].into(),
+                block_commitments_hash,
+                light_client_root_hash: block_commitments_hash,
+                final_sapling_root_hash: block_commitments_hash,
                 default_roots: DefaultRoots {
                     merkle_root,
-                    chain_history_root: [0; 32].into(),
+                    chain_history_root,
```

Comment:
Can you check if this is the same hash as the one returned by `zcashd`?

I'm not sure if the history tree in our database is the one used in the `tip_height` block commitment, or the one after the changes in the `tip_height` block have been applied to the history tree.

You could also check the state RFC or state commit task code to see what it does.

Reply:
@oxarbitrage can you please check if you need to add the tip block to the chain history root, or if it has already been added by the state?

Reply:
If the chain history hash matches zcashd, and they are at the same tip height, then we've got it correct. So you could test first, then look it up if the test fails.

Reply:
Resolved by testing against `zcashd`:
https://github.com/ZcashFoundation/zebra/pull/5751#issuecomment-1339602554